src/status_im/chat/models/message.cljs

```diff
@@ -79,19 +79,14 @@
       (requests-events/add-request chat-identifier enriched-message))))
 
 (defn add-to-chat?
-  [{:keys [db get-stored-message]} {:keys [group-id chat-id from message-id]}]
+  [{:keys [db]} {:keys [group-id chat-id from message-id]}]
   (let [chat-identifier                                  (or group-id chat-id from)
         {:keys [chats deleted-chats current-public-key]} db
         {:keys [messages not-loaded-message-ids]}        (get chats chat-identifier)]
     (when (not= from current-public-key)
-      (if group-id
-        (not (or (get deleted-chats chat-identifier)
-                 (get messages message-id)
-                 (get not-loaded-message-ids message-id)))
-        (not (or (get messages message-id)
-                 (get not-loaded-message-ids message-id)
-                 (and (get deleted-chats chat-identifier)
-                      (get-stored-message message-id))))))))
+      (not (or (get deleted-chats chat-identifier)
+               (get messages message-id)
+               (get not-loaded-message-ids message-id))))))
```

Comment:
I'm afraid we can't get rid of the realm message read in this case, the following scenario is problematic:
You have some deleted chat and message for that chat get re-delivered to you (because you didn't ack before, or any other reason, I generally found the re-delivery quite unreliable and you always how to defend against message with same id delivered to you again).
Now we now that the message is for deleted chat (as deleted chat ids are loaded on startup) but we don't know if it's message not received before (in which case, we have to restore the chat and add message, otherwise once you delete 1-1 chat, you will never be able to have conversation with given contact anymore) or already received message (so nothing is restored and we ignore the message).
The reason for it, is that for deleted chats, we don't load their message ids into db at startup.

I did it that way originally because:
1. It's a quite rare case (re-delivery of message for deleted chat) so the perf impact of reading from realm (only in that specific case) is probably negligible overall
2. Having `:not-loaded-message-ids` for deleted chats always in db seemed like a waste

Maybe point 2) is open to discussion and it's nothing horrible.

Reply:
@janherich good call. Returned that logic back.